### DISCUSSION TITLE
Problems setting up VirtualBox with SystemD

### DISCUSSION THREAD

**Discussion by VolvoBirkenstock** [mocaccinoOS/mocaccino#113] [General] [answered]
Hi, Mocaccino Gurus. 
I have come to realize how dependent I became on the Sabayon HOWTO pages. :-)  I have forgotten a lot about how to install VirtualBox and setup proper operation with systemd.  I have used luet to install apps/virtualbox, kernel-modules/virtualbox-guest-additions-lts, and kernel-modules/virtualbox-modules-lts in the hope that this would provide all the software needed.  (These three items were returned when I executed "luet search virtualbox".)  I seem to recall that after installing VirtualBox in Sabayon, I would see the following: vboxdrv, vboxnetadp, and vboxnetflt.  I also think that these names had to be entered into a file called /etc/modules-load.d/virtualbox.conf or something similar.  I think we could then initialize this with systemd by running a command similar to 
`systemctl start systemd-modules-load`
I have performed 
`find /etc -iname "*vboxdrv*" -print`
which returned
`/etc/modprobe.d/vboxdrv.conf`
The contents of /etc/modprobe.d/vboxdrv.conf are all commented out. 
I have also performed 
`find /etc -iname "*vboxnetadp*" -print`
`find /etc -iname "*vboxnetflt*" -print`
but they return nothing. 

When I try to start VirtualBox using the desktop icon, I obtain a popup saying "Failed to open a session for the virtual machine <MachineName>".  I clicked on the Details.  It reports 
> Kernel driver not installed (rc=-1908) 

along with the statement 

> The VirtualBox Linux kernel driver is either not loaded or not set up correctly. Please try setting it up again by executing '/sbin/vboxconfig' as root.  
> If your system has EFI Secure Boot enabled you may also need to sign the kernel modules (vboxdrv, vboxnetflt, vboxnetadp, vboxpci) before you can load them. Please see your Linux system's documentation for more information.  
> where: suplibOsInit what: 3 VERR_VM_DRIVER_NOT_INSTALLED (-1908) - The support driver is not installed. On linux, open returned ENOENT. 

Executing /sbin/vboxconfig fails to work since the vboxconfig file does not exist anywhere on the system. 

Can you tell me how to correct this? 
Thank you very much in advance.

**Comment by joostruis** [comment, marked as answer]
Have you tried: modprobe vboxdrv ?

**Reply by VolvoBirkenstock** [reply]
I am sorry, @joostruis , but I no longer have Mocaccino installed on hardware.  Since I asked the above question I have reverted the hardware to another OS and installed Mocaccino into a VM.  I do not believe that I can run virtualbox inside a VM, but I could be wrong.  (A VM inside a VM?)  I have tested Octave inside the Mocaccino VM and it worked well.  I hope to eventually load Mocaccino onto hardware again because I like the Mocaccino/Sabayon/Gentoo model.
